logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java

```diff
@@ -0,0 +1,652 @@
+        return enabledBuiltIns.contains(requireNonNull(property, "property"));
+    }
+
+    public Set<BuiltInProperty> getBuiltInProperties() {
+        return enabledBuiltIns.toSet();
+    }
+
+    public void addAttribute(String alias, AttributeKey<?> attrKey) {
+        requireNonNull(attrKey, "attrKey");
+        requireNonNull(alias, "alias");
+        attrNames.put(attrKey, "attrs." + alias);
+    }
+
+    public boolean containsAttribute(AttributeKey<?> key) {
+        requireNonNull(key, "key");
+        return attrNames.containsKey(key);
+    }
+
+    public Map<AttributeKey<?>, String> getAttributes() {
+        return Collections.unmodifiableMap(attrNames.entrySet().stream().collect(
+                Collectors.toMap(Entry::getKey, e -> e.getValue().substring(6))));
+    }
+
+    public void addHttpRequestHeader(CharSequence name) {
+        addHttpHeader("req.http_headers", httpReqHeaderNames, name);
+    }
+
+    public void addHttpResponseHeader(CharSequence name) {
+        addHttpHeader("res.http_headers", httpResHeaderNames, name);
+    }
+
+    private static void addHttpHeader(
+            String mdcKeyPrefix, Map<AsciiString, String> httpHeaderNames, CharSequence name) {
+        final AsciiString key = toHeaderName(name);
+        final String value = mdcKeyPrefix + '.' + key;
+        httpHeaderNames.put(key, value);
+    }
+
+    public boolean containsHttpRequestHeader(CharSequence name) {
+        return httpReqHeaderNames.containsKey(toHeaderName(name));
+    }
+
+    public boolean containsHttpResponseHeader(CharSequence name) {
+        return httpResHeaderNames.containsKey(toHeaderName(name));
+    }
+
+    private static AsciiString toHeaderName(CharSequence name) {
+        return HttpHeaderNames.of(requireNonNull(name, "name").toString());
+    }
+
+    public Set<AsciiString> getHttpRequestHeaders() {
+        return Collections.unmodifiableSet(httpReqHeaderNames.keySet());
+    }
+
+    public Set<AsciiString> getHttpResponseHeaders() {
+        return Collections.unmodifiableSet(httpResHeaderNames.keySet());
+    }
+
+    public void setExport(String mdcKey) {
+        requireNonNull(mdcKey, "mdcKey");
+
+        final Optional<BuiltInProperty> opt = BuiltInProperty.findByMdcKey(mdcKey);
+        if (opt.isPresent()) {
+            enabledBuiltIns.add(opt.get());
+            return;
+        }
+
+        if (mdcKey.startsWith("attrs.")) {
+            final int colonIdx = mdcKey.indexOf(':');
+            if (colonIdx < 0) {
+                throw new IllegalArgumentException(
+                        "invalid MDC key: " + mdcKey + " (expected: attrs.<alias>:<AttributeKey.name>)");
+            }
+
+            final String alias = mdcKey.substring(6, colonIdx);
+            final AttributeKey<?> attrKey = AttributeKey.valueOf(mdcKey.substring(colonIdx + 1));
+            addAttribute(alias, attrKey);
+            return;
+        }
+
+        if (mdcKey.startsWith("req.http_headers.")) {
```

Comment:
How about define string constant and use it? 
`mdcKey.substring(17)` is hard to understand for me.